blk/components/blk_components.mk

```diff
@@ -0,0 +1,39 @@
+#
+# Copyright 2024, UNSW
+#
+# SPDX-License-Identifier: BSD-2-Clause
+#
+# This Makefile snippet builds the blk RX and TX virtualisers
+# it should be included into your project Makefile
+#
+# NOTES:
+#  Generates blk_virt.elf
+#
+
+
+BLK_IMAGES := blk_virt.elf
+
+CFLAGS_blk := -I ${SDDF}/include
+
+CHECK_BLK_FLAGS_MD5:=.blk_cflags-$(shell echo -- ${CFLAGS} ${CFLAGS_blk} | shasum | sed 's/ *-//')
+
+${CHECK_BLK_FLAGS_MD5}:
+	-rm -f .blk_cflags-*
+	touch $@
+
+
+blk_virt.elf: virt.o
+	$(LD) $(LDFLAGS) $^ $(LIBS) -o $@
+
+virt.o: ${CHECK_BLK_FLAGS_MD5}
+virt.o: ${SDDF}/blk/components/virt.c
+	${CC} ${CFLAGS} ${CFLAGS_blk} -o $@ -c $<
+
+clean::
+	rm -f blk_virt.[od] .blk_cflags-*
+
+clobber::
+	rm -f ${BLK_IMAGES}
+
+
+-include virt.d
```

Comment:
Should this be `blk_virt.d`?

Reply:
It was technically correct as `virt.d` since the `.o` file was called `virt.o` but I've changed them both to `blk_virt` to avoid conflicts. May be worth also doing this for the serial virt in another PR:
https://github.com/au-ts/sddf/blob/main/serial/components/serial_components.mk

Reply:
Ah right. I mentioned it because the `clean` was doing `blk_virt.[od]`

Reply:
Ah I see